ironfish/src/rpc/routes/mempool/getTransactions.ts

```diff
@@ -0,0 +1,137 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { getFeeRate, MemPool } from '../../../memPool'
+import { Transaction } from '../../../primitives'
+import { ApiNamespace, router } from '../router'
+
+type MinMax = {
+  min?: number
+  max?: number
+}
+
+export type GetMempoolTransactionsRequest = {
+  limit?: number
+  feeRate?: MinMax
+  fee?: MinMax
+  expiration?: MinMax
+  position?: MinMax
+  stream?: boolean
+}
+
+export type GetMempoolTransactionResponse = {
+  serializedTransaction: string
+  position: number
+}
+
+export type GetMempoolTransactionsResponse = {
+  transactions: GetMempoolTransactionResponse[]
+}
+
+const minMaxSchema = yup.object({
+  min: yup.number().optional().min(0),
+  max: yup.number().optional(),
+})
+
+export const MempoolTransactionsRequestSchema: yup.ObjectSchema<GetMempoolTransactionsRequest> =
+  yup
+    .object({
+      limit: yup.number().min(0),
+      feeRate: minMaxSchema.optional(),
+      fee: minMaxSchema.optional(),
+      expiration: minMaxSchema.optional(),
+      position: minMaxSchema.optional(),
+      stream: yup.boolean().optional(),
+    })
+    .required()
+    .defined()
+
+export const MempoolTransactionsResponseSchema: yup.ObjectSchema<GetMempoolTransactionsResponse> =
+  yup
+    .object({
+      transactions: yup
+        .array()
+        .of(
+          yup
+            .object({
+              serializedTransaction: yup.string().defined(),
+              position: yup.number().defined(),
+            })
+            .defined(),
+        )
+        .defined(),
+    })
+    .defined()
+
+router.register<typeof MempoolTransactionsRequestSchema, GetMempoolTransactionsResponse>(
+  `${ApiNamespace.mempool}/getTransactions`,
+  MempoolTransactionsRequestSchema,
+  (request, node): void => {
+    if (!request.data.stream) {
```

Comment:
I don't think we should implement this API, it should always be streaming.

Reply:
Yea I had the same concerns about having a large mempool. The main reason I kept it was that miners already run this function about once every minute to create new blocks. Running it once every 5 seconds is adding load but I thought it was reasonably similar to mining load to be possible.

Streaming is nice to have but not necessary for debugging use cases. Maybe we could implement a max mempool size to stream? Streaming 500tx mempool is still pretty fast but maybe we just disallow streaming with greater mempool sizes. Or I can remove the streaming all-together